source/controller/Award.ts

```diff
@@ -0,0 +1,135 @@
+import {
+    Authorized,
+    Body,
+    CurrentUser,
+    Delete,
+    Get,
+    HttpCode,
+    JsonController,
+    NotFoundError,
+    OnNull,
+    OnUndefined,
+    Param,
+    Patch,
+    Post
+} from 'routing-controllers';
+import { ResponseSchema } from 'routing-controllers-openapi';
+
+import { dataSource, Hackathon, User, Award, Media } from '../model';
+
+import { HackathonController } from './Hackathon';
+import { ActivityLogController } from './ActivityLog';
+
+@JsonController('/hackathon/:hackathonName/award')
+export class AwardController {
+    store = dataSource.getRepository(Award);
+    hackathonStore = dataSource.getRepository(Hackathon);
+
+    @Post()
+    @Authorized()
+    @HttpCode(201)
+    @ResponseSchema(Award)
+    async createOne(
+        @CurrentUser() currentUser: User,
+        @Param('hackathonName') hackathonName: string,
+        @Body()
+        awardData: {
+            name: string;
+            description: string;
+            quantity: number;
+            target: 'team' | 'individual';
+            pictures: Media[];
+        }
```

Comment:
```suggestion
        awardData: Award
```
 `class` 才既有类型信息，又有运行时的实体，才能在服务器启动的时候生成 Swagger 文档。